src/ethereum/ethEngine.js

```diff
@@ -588,11 +587,7 @@ export class EthereumEngine extends CurrencyEngine {
           throw e
         }
       } else {
-        nonce = this.walletLocalData.otherData.nextNonce
-        this.walletLocalData.otherData.unconfirmedNextNonce = bns.add(
-          this.walletLocalData.otherData.nextNonce,
-          '1'
-        )
+        nonce = otherParams.nonceArg
```

Comment:
Isn't this a noop because of lines 558 and 559? This is a clue to me that perhaps the issue with the code changed in this WIP PR.